src/service/ampdoc-impl.js

```diff
@@ -223,6 +285,30 @@ export class AmpDocSingle extends AmpDoc {
   getUrl() {
     return this.win.location.href;
   }
+
+  /** @override */
+  getBody() {
+    return this.win.document.body;
+  }
+
+  /** @override */
+  onBody(callback) {
+    if (this.win.document.body) {
+      callback(this.win.document.body);
```

Comment:
Actually I think you are right. It's now a good time as any to fix this issue. Let me see if I can come up with a better API to support a quick access in most of cases and async when the state is not available.